accounts/abi/bind/base.go

```diff
@@ -223,7 +227,7 @@ func (c *BoundContract) transact(opts *TransactOpts, contract *common.Address, i
 	}
 	// Figure out the gas allowance and gas price values
 	gasPrice := opts.GasPrice
-	if gasPrice == nil {
+	if gasPrice == nil && opts.FeeCapPerGas == nil && opts.MaxMinerBribePerGas == nil {
```

Comment:
Suggest to assign `opts.FeeCapPerGas` and `opts.MaxMinerBribePerGas` to variables like `gasPrice := opts.GasPrice` because they are used in later code:
```
feeCapPerGas := opts.FeeCapPerGas
maxMinerBribePerGas := opts.MaxMinerBribePerGas
```